frontend/src/lib/components/Errors/ErrorDisplay.tsx

```diff
@@ -2,66 +2,112 @@ import './ErrorDisplay.scss'
-    type: 'raw'
-    frames: StackFrame[]
-}
-interface ResolvedStackTrace {
-    type: 'resolved'
-    frames: StackFrame[]
-}
+function StackTrace({
+    frames,
+    showAllFrames,
+}: {
+    frames: ErrorTrackingStackFrame[]
+    showAllFrames: boolean
+}): JSX.Element | null {
+    const { frameContexts } = useValues(stackFrameLogic)
+    const { loadFrameContexts } = useActions(stackFrameLogic)
+    const displayFrames = showAllFrames ? frames : frames.filter((f) => f.in_app)
 
-interface Exception {
-    stacktrace: ResolvedStackTrace | RawStackTrace
-    module: string
-    type: string
-    value: string
-}
+    useEffect(() => {
+        loadFrameContexts({ frames })
+    }, [frames, loadFrameContexts])
 
-function StackTrace({ frames, showAllFrames }: { frames: StackFrame[]; showAllFrames: boolean }): JSX.Element | null {
-    const displayFrames = showAllFrames ? frames : frames.filter((f) => f.in_app)
+    const initiallyActiveIndex = displayFrames.findIndex((f) => f.in_app) || 0
 
-    const panels = displayFrames.map(({ filename, lineno, colno, function: functionName }, index) => {
+    const panels = displayFrames.map(({ raw_id, source, line, column, resolved_name, lang }, index) => {
+        const frameContext = frameContexts[raw_id]
         return {
             key: index,
             header: (
                 <div className="flex flex-wrap space-x-0.5">
-                    <span>{filename}</span>
-                    {functionName ? (
+                    <span>{source}</span>
+                    {resolved_name ? (
                         <div className="flex space-x-0.5">
                             <span className="text-muted">in</span>
-                            <span>{functionName}</span>
+                            <span>{resolved_name}</span>
                         </div>
                     ) : null}
-                    {lineno && colno ? (
+                    {line ? (
                         <div className="flex space-x-0.5">
                             <span className="text-muted">at line</span>
                             <span>
-                                {lineno}:{colno}
+                                {line}
+                                {column && `:${column}`}
                             </span>
                         </div>
                     ) : null}
                 </div>
             ),
-            content: null,
+            content: frameContext ? <FrameContext context={frameContext} language={getLanguage(lang)} /> : null,
+            className: 'p-0',
         }
     })
 
-    return <LemonCollapse defaultActiveKeys={[]} multiple panels={panels} size="xsmall" />
+    return <LemonCollapse defaultActiveKeys={[initiallyActiveIndex]} multiple panels={panels} size="xsmall" />
+}
+
+function FrameContext({
+    context,
+    language,
+}: {
+    context: ErrorTrackingStackFrameContext
+    language: Language
+}): JSX.Element {
+    const { before, line, after } = context
```

Comment:
I think we should sort `before` and `after` by line number before displaying them, I don't want "the context has to be sorted in the DB" to be an invariant we have to carefully maintain in cymbal (I don't expect we'll change how we construct the context object for javascript, but who knows what other languages will bring).